lib/src/rules/prefer_relative_imports.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/file_system/file_system.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/ast.dart';
+import 'package:linter/src/rules/implementation_imports.dart';
+import 'package:yaml/yaml.dart';
+
+const _desc = r'Prefer relative imports for files in `lib/`.';
+
+const _details = r'''Prefer relative imports for files in `lib/`.
+
+When mixing relative and absolute imports it's possible to create confusion
+where the same member gets imported in two different ways. One way to avoid
+that is to ensure you consistently use relative imports for files withing the
+`lib/` directory.
+
+**GOOD:**
+
+```
+import 'bar.dart';
+...
```

Comment:
Personally I'd remove the "..." (here and below) because I don't think it adds to the example. Your call.